umbraco/Umbraco-CMS
# V14: Return all (even nested) compositions when getting a content type by id

## Details
- GET <code>/umbraco/management/api/v1/<b>document-type</b>/{id}</code>, <code>/umbraco/management/api/v1/<b>media-type</b>/{id}</code> and <code>/umbraco/management/api/v1/<b>member-type</b>/{id}</code> endpoints included only the direct compositions in their response:
  - Considering the following structure:
    - _Composition - 27f0144b-644d-47c9-a17e-47bcf1caaea0_
      - _Child composition - dfb90df0-53db-4ea8-8cc0-058d7f3cd1e2_
        - _Grandchild composition - 7c39cd33-30f5-4bf2-b686-fd35e772c559_
  - GET <code>/umbraco/management/api/v1/document-type/<b>7c39cd33-30f5-4bf2-b686-fd35e772c559</b></code> returned:
```json
  . . .
  "compositions": [
    {
      "documentType": {
        "id": "dfb90df0-53db-4ea8-8cc0-058d7f3cd1e2"
      },
      "compositionType": "Inheritance"
    }
  ]	
```
- This PR retrieves all compositions - direct and all ancestor ones - and includes them in the response of those endpoints.

## Test
- Create a similar document/media/member type structure:
  - _Composition - 27f0144b-644d-47c9-a17e-47bcf1caaea0_
    - _Child composition - dfb90df0-53db-4ea8-8cc0-058d7f3cd1e2_
      - _Grandchild composition - 7c39cd33-30f5-4bf2-b686-fd35e772c559_
- Use GET <code>/umbraco/management/api/v1/<b>document-type</b>/{id}</code>, <code>/umbraco/management/api/v1/<b>media-type</b>/{id}</code> and <code>/umbraco/management/api/v1/<b>member-type</b>/{id}</code> 
- Verify that the `"compositions"` collection of each endpoint contains the correct compositions data;
  - Pay attention to the value `"compositionType"` - when adding another composition (not through inheritance) - it would be added to the collection with `"compositionType": "Composition"`
    - Verify that this is the case
